pkg/cmd/pr/checks/collect.go

```diff
@@ -0,0 +1,94 @@
+package checks
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/cli/cli/v2/pkg/cmd/pr/shared"
+)
+
+type check struct {
+	Name        string    `json:"name"`
+	State       string    `json:"state"`
+	StartedAt   time.Time `json:"startedAt"`
+	CompletedAt time.Time `json:"completedAt"`
+	Link        string    `json:"link"`
+	Bucket      string    `json:"bucket"`
+}
+
+type checksMeta struct {
+	Failed   int
+	Passed   int
+	Pending  int
+	Skipping int
+}
+
+func collect(opts *ChecksOptions) ([]check, checksMeta, error) {
```

Comment:
Right now it feels like this function is doing too much work. I think we should separate out the functionality of retrieving the checks from the API and aggregation of the checks into the `checks` and `meta` structs. Additionally, `collect` does not feel descriptive enough to me what do you think of `aggregateChecks` or `categorizeChecks`?